src/test/java/org/codewars/utils/ReaderTest.java

```diff
@@ -73,6 +74,32 @@ public void testReadFloatInv(String input, float expected, String expectedOut) {
 
     @Test
     public void testReadLong() {
+        System.setIn(new ByteArrayInputStream("9876543210".getBytes()));
+        Reader reader = new Reader();
+        long actual = reader.readLong();
+        Assert.assertEquals(actual, 9876543210L);
+    }
+
+    @DataProvider(name = "readLongTestDataInvalid")
+    private Object[][] readLongTestDataInv() {
+        return new Object[][]{
+                {"qwerty\nqwertyqwerty\n1234567890", 1234567890L, "Your value is invalid. Try again\nYour value is invalid. Try again\n"},
+                {"qwerty\n1234567890", 1234567890L, "Your value is invalid. Try again\n"},
+        };
+    }
+
+    @Test(dataProvider = "readLongTestDataInvalid")
+    public void testReadLongInvalid(String input, long expected, String expectedOut) {
+        System.setIn(new ByteArrayInputStream(input.getBytes()));
+        OutputStream out = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(out));
+        Reader reader = new Reader();
+        try {
+            long actual = reader.readLong();
+            Assert.assertEquals(actual, expected);
+        } catch (NoSuchElementException e) {
+            Assert.assertEquals(e.getMessage(), expectedOut);
+        }
```

Comment:
`public long readLong() {
        while (true) {
            try {
                return scanner.nextLong();
            } catch (Exception e) {
                System.out.println("Your value is invalid. Try again");
                scanner.nextLine();
            }
        }
    }`
 here'll be console msg and jne more try to get Long instead of exception

Reply:
rermove try/catch? and add smth like this:

```
// Check if the console message matches the expected output??
    Assert.assertEquals(out.toString().replace("\r", ""), expectedOut);

    Assert.assertEquals(actual, expected);
```